packages/ozone/src/verification/service.ts

```diff
@@ -0,0 +1,192 @@
+import { Selectable } from 'kysely'
+import {
+  $Typed,
+  AppBskyActorDefs,
+  AtUri,
+  ToolsOzoneModerationDefs,
+  ToolsOzoneVerificationDefs,
+} from '@atproto/api'
+import { Database } from '../db'
+import { CreatedAtUriKeyset, paginate } from '../db/pagination'
+import { Verification } from '../db/schema/verification'
+
+export type VerificationServiceCreator = (db: Database) => VerificationService
+
+export class VerificationService {
+  constructor(public db: Database) {}
+
+  static creator() {
+    return (db: Database) => new VerificationService(db)
+  }
+
+  async create(
+    verifications: Pick<
+      Verification,
+      'uri' | 'issuer' | 'subject' | 'handle' | 'displayName' | 'createdAt'
+    >[],
+  ) {
+    // @TODO: We should probably handle errors here?
+    await this.db.transaction(async (tx) => {
+      return Promise.allSettled(
+        verifications.map((verification) => {
+          return tx.db
+            .insertInto('verification')
+            .values(verification)
+            .onConflict((oc) => oc.doNothing())
+            .execute()
+        }),
+      )
+    })
+  }
+
+  async markRevoked({
+    uris,
+    revokedBy,
+    revokedAt,
+    revokeReason,
+  }: {
+    uris: string[]
+    revokedBy?: string
+    revokedAt?: string
+    revokeReason?: string
+  }) {
+    const now = new Date().toISOString()
+    return Promise.allSettled(
+      uris.map(async (uri) => {
```

Comment:
It looks like callers of `markRevoked()` are ignoring errors surfaced here.  Perhaps it's more straightforward to use a transaction and allow all the updates to fail or succeed together, then surface the error if they fail.

Reply:
yh definitely easier. i was thinking we could allow partial success but makes things easier to just fail the entire transaction